pam/internal/adapter/model.go

```diff
@@ -257,6 +257,29 @@ func (m *UIModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 			m.gdmModel, gdmCmd = m.gdmModel.Update(msg)
 		}
 
+		// If the mode is passwd (meaning we are trying to change the user password) and we already have a AUTHTOK, we
+		// don't need to ask the user for a new one.
+		if msg.layout.Type == "newpassword" && m.SessionMode == authd.SessionMode_PASSWD {
+			challenge, err := m.PamMTx.GetItem(pam.Authtok)
+			if err == nil && challenge != "" {
+				cmds := []tea.Cmd{
+					m.authenticationModel.Compose(
+						m.currentSession.brokerID,
+						m.currentSession.sessionID,
+						m.currentSession.encryptionKey,
+						msg.layout,
+					),
```

Comment:
This command is the same we're sending in the default mode, since it's quite complex per se already, I'd define it just once before entering here and then adding it to the slice or just returning it as we already do.

Reply:
Good spot!